{"name": "test_sending", "status": "failed", "statusDetails": {"message": "selene.core.exceptions.TimeoutException: Message: \n\nTimed out after 3.0s, while waiting for:\nbrowser.all(('css selector', '.table-responsive td:nth-child(2)')).has texts (Student(first_name='Anna', last_name='Vishnyakova', email='mypochta@pochta.ru', phone='89000000000', address='Москва', birthday=datetime.date(1994, 6, 3), hobby=[<Hobby.Sports: 'Sports'>, <Hobby.Music: ('Music',)>], image='test.png', gender=<Gender.Female: ('Female',)>, subjects=[<Subject.Maths: ('Maths',)>, <Subject.Arts: ('Arts',)>], state=<State.Haryana: ('Haryana',)>, city=<City.Panipat: ('Panipat',)>),)\n\nReason: AssertionError: actual visible_texts: ['Anna Vishnyakova', 'mypochta@pochta.ru', 'Female', '8900000000', '03 July,1994', 'Maths, Arts', 'Sports, Music', 'test.png', 'Москва', 'Haryana Panipat']\nScreenshot: file:///Users/annavishnyakova/.selene/screenshots/1674320245934/1674320245935.png\nPageSource: file:///Users/annavishnyakova/.selene/screenshots/1674320245934/1674320245935.html", "trace": "fn = <selene.core.conditions.CollectionCondition object at 0x10f6f9c90>\n\n    @self._decorator(self)\n    def _(fn: Callable[[E], R]) -> R:\n        finish_time = time.time() + self._timeout\n    \n        while True:\n            try:\n>               return fn(self.entity)\n\nvenv/lib/python3.10/site-packages/selene/core/wait.py:122: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selene.core.conditions.CollectionCondition object at 0x10f6f9c90>\nentity = <selene.core.entity.Collection object at 0x10f6f9270>\n\n    def __call__(self, entity: E) -> None:\n>       return self._fn(entity)\n\nvenv/lib/python3.10/site-packages/selene/core/condition.py:174: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nentity = <selene.core.entity.Collection object at 0x10f6f9270>\n\n    def fn(entity: E) -> None:\n        query_to_str = str(query)\n        result = (\n            query.__name__\n            if query_to_str.startswith('<function')\n            else query_to_str\n        )\n        actual = query(entity)\n        if not predicate(actual):\n>           raise AssertionError(f'actual {result}: {actual}')\nE           AssertionError: actual visible_texts: ['Anna Vishnyakova', 'mypochta@pochta.ru', 'Female', '8900000000', '03 July,1994', 'Maths, Arts', 'Sports, Music', 'test.png', 'Москва', 'Haryana Panipat']\n\nvenv/lib/python3.10/site-packages/selene/core/condition.py:147: AssertionError\n\nDuring handling of the above exception, another exception occurred:\n\n    def test_sending():\n        a_vishnyakova = Student(\n            first_name='Anna',\n            last_name='Vishnyakova',\n            email='mypochta@pochta.ru',\n            phone='89000000000',\n            address='Москва',\n            birthday=date(1994, 6, 3),\n            gender=Gender.Female,\n            subjects=[Subject.Maths, Subject.Arts],\n            hobby=[Hobby.Sports, Hobby.Music],\n            image='test.png',\n            state=State.Haryana,\n            city=City.Panipat\n        )\n    \n        practice_form.open()\n        practice_form.fill_data(a_vishnyakova)\n        practice_form.submit()\n    \n>       practice_form.validation_data(a_vishnyakova)\n\ntests/test_form_sending.py:25: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \ndemoqa_tests_selene/model/pages/practice_form.py:65: in validation_data\n    modal.check_form('.table-responsive td:nth-child(2)', *value)\ndemoqa_tests_selene/model/controls/modal.py:6: in check_form\n    browser.all(form).should(have.texts(*value))\nvenv/lib/python3.10/site-packages/selene/core/entity.py:1088: in should\n    self.wait.for_(condition)\nvenv/lib/python3.10/site-packages/selene/core/wait.py:147: in for_\n    return _(fn)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nfn = <selene.core.conditions.CollectionCondition object at 0x10f6f9c90>\n\n    @self._decorator(self)\n    def _(fn: Callable[[E], R]) -> R:\n        finish_time = time.time() + self._timeout\n    \n        while True:\n            try:\n                return fn(self.entity)\n            except Exception as reason:\n                if time.time() > finish_time:\n    \n                    reason_message = str(reason)\n    \n                    reason_string = '{name}: {message}'.format(\n                        name=reason.__class__.__name__,\n                        message=reason_message,\n                    )\n                    # todo: think on how can we improve logging failures in selene, e.g. reverse msg and stacktrace\n                    # stacktrace = getattr(reason, 'stacktrace', None)\n                    timeout = self._timeout\n                    entity = self.entity\n    \n                    failure = TimeoutException(\n                        f'\\n'\n                        f'\\nTimed out after {timeout}s, while waiting for:'\n                        f'\\n{entity}.{fn}'\n                        f'\\n'\n                        f'\\nReason: {reason_string}'\n                    )\n    \n>                   raise self._hook_failure(failure)\nE                   selene.core.exceptions.TimeoutException: Message: \nE                   \nE                   Timed out after 3.0s, while waiting for:\nE                   browser.all(('css selector', '.table-responsive td:nth-child(2)')).has texts (Student(first_name='Anna', last_name='Vishnyakova', email='mypochta@pochta.ru', phone='89000000000', address='Москва', birthday=datetime.date(1994, 6, 3), hobby=[<Hobby.Sports: 'Sports'>, <Hobby.Music: ('Music',)>], image='test.png', gender=<Gender.Female: ('Female',)>, subjects=[<Subject.Maths: ('Maths',)>, <Subject.Arts: ('Arts',)>], state=<State.Haryana: ('Haryana',)>, city=<City.Panipat: ('Panipat',)>),)\nE                   \nE                   Reason: AssertionError: actual visible_texts: ['Anna Vishnyakova', 'mypochta@pochta.ru', 'Female', '8900000000', '03 July,1994', 'Maths, Arts', 'Sports, Music', 'test.png', 'Москва', 'Haryana Panipat']\nE                   Screenshot: file:///Users/annavishnyakova/.selene/screenshots/1674320245934/1674320245935.png\nE                   PageSource: file:///Users/annavishnyakova/.selene/screenshots/1674320245934/1674320245935.html\n\nvenv/lib/python3.10/site-packages/selene/core/wait.py:145: TimeoutException"}, "start": 1674320245969, "stop": 1674320277025, "uuid": "af164614-e306-4391-aece-e365d430df4e", "historyId": "f84a44ba9169ec3b67bc3bab3111c34d", "testCaseId": "404defe7aa34c817caa299928647ea50", "fullName": "tests.test_form_sending#test_sending", "labels": [{"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_form_sending"}, {"name": "host", "value": "MacBook-Pro-Anna.local"}, {"name": "thread", "value": "83000-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_form_sending"}]}